tests/unit_tests/test_wave_06.py

```diff
@@ -0,0 +1,227 @@
+import pytest
+from swap_meet.vendor import Vendor
+from swap_meet.clothing import Clothing
+from swap_meet.decor import Decor
+from swap_meet.electronics import Electronics
+
+
+def test_best_by_category():
+    item_a = Clothing(condition=2.0)
+    item_b = Decor(condition=2.0)
+    item_c = Clothing(condition=4.0)
+    item_d = Decor(condition=5.0)
+    item_e = Clothing(condition=3.0)
+    tai = Vendor(
+        inventory=[item_a, item_b, item_c, item_d, item_e]
+    )
+
+    best_item = tai.get_best_by_category("Clothing")
+
+    assert best_item.category == "Clothing"
+    assert best_item.condition == pytest.approx(4.0)
+
+def test_best_by_category_no_matches_is_none():
+    item_a = Decor(condition=2.0)
+    item_b = Decor(condition=2.0)
+    item_c = Decor(condition=4.0)
+    tai = Vendor(
+        inventory=[item_a, item_b, item_c]
+    )
+
+    best_item = tai.get_best_by_category("Electronics")
+
+    assert best_item is None
+
+def test_best_by_category_with_duplicates():
+    # Arrange
+    item_a = Clothing(condition=2.0)
+    item_b = Clothing(condition=4.0)
+    item_c = Clothing(condition=4.0)
+    tai = Vendor(
+        inventory=[item_a, item_b, item_c]
+    )
+
+    # Act
+    best_item = tai.get_best_by_category("Clothing")
+
+    # Assert
+    assert best_item.category == "Clothing"
+    assert best_item.condition == pytest.approx(4.0)
+
+def test_swap_best_by_category():
+    # Arrange
+    # me
+    item_a = Decor(condition=2.0)
+    item_b = Electronics(condition=4.0)
+    item_c = Decor(condition=4.0)
+    tai = Vendor(
+        inventory=[item_a, item_b, item_c]
+    )
+
+    # them
+    item_d = Clothing(condition=2.0)
+    item_e = Decor(condition=4.0)
+    item_f = Clothing(condition=4.0)
+    jesse = Vendor(
+        inventory=[item_d, item_e, item_f]
+    )
+
+    # Act
+    result = tai.swap_best_by_category(
+        other=jesse,
+        my_priority="Clothing",
+        their_priority="Decor"
+    )
+
+    # *********************************************************************
+    # ****** Complete Assert Portion of this test **********
+    # *********************************************************************
+    # Assertions should check:
+    # - That the results is truthy
+    # - That tai and jesse's inventories are the correct length
+    # - That all the correct items are in tai and jesse's inventories, and that the items that were swapped are not there
```

Comment:
It's a reach, but I had the thought that the tail end of the sentence could possibly give the impression those items shouldn't be in either vendor, feel free to ignore if it doesn't feel necessary. 

```suggestion
    # - That all the correct items are in tai and jesse's inventories, including the items which were swapped from one vendor to the other
```